src/main/scala/com/cognite/sdk/scala/v1/resources/sequenceRows.scala

```diff
@@ -55,50 +57,112 @@ class SequenceRows[F[_]](val requestSession: RequestSession[F])
         _ => ()
       )
 
-  def queryById(
-      id: Long,
-      inclusiveStart: Long,
-      exclusiveEnd: Long,
-      limit: Option[Int] = None,
-      columns: Option[Seq[String]] = None
-  ): F[(Seq[SequenceColumnId], Seq[SequenceRow])] =
+  private def sendQuery(query: SequenceRowsQuery, batchSize: Int) =
     requestSession
-      .post[(Seq[SequenceColumnId], Seq[SequenceRow]), SequenceRowsResponse, SequenceRowsQueryById](
-        SequenceRowsQueryById(id, inclusiveStart, exclusiveEnd, limit, columns),
+      .post[SequenceRowsResponse, SequenceRowsResponse, SequenceRowsQuery](
+        query.withCursorAndLimit(
+          query.cursor,
+          Some(math.min(batchSize, query.limit.getOrElse(batchSize)))
+        ),
         uri"$baseUrl/list",
-        value => (value.columns.toList, value.rows)
+        (v: SequenceRowsResponse) => v
       )
 
+  private def pullQueryResults(
+      query: SequenceRowsQuery,
+      batchSize: Int
+  ): fs2.Stream[F, SequenceRowsResponse] =
+    Readable.pageThroughCursors[F, Option[Int], SequenceRowsResponse](
+      query.cursor,
+      query.limit,
+      (cursor: Option[String], limit: Option[Int]) =>
+        if (limit.exists(_ <= 0)) {
+          F.pure(None: Option[(SequenceRowsResponse, Option[Int])])
```

Comment:
If that's what we do for others, I guess. I think we should maybe throw an exception if this happens, rather than silently accept a bad parameter.

Reply:
There's also `Option.empty[(SequenceRowsResponse, Option[Int])]` which is arguably nicer than having to cast.

Reply:
>If that's what we do for others, I guess.

Yes, we do that for other resources. It's there mostly to stop the recursion.